Shape_detection/include/CGAL/Shape_detection/Region_growing/Region_growing_on_segment_set/Least_squares_line_fit_region.h

```diff
@@ -0,0 +1,334 @@
+    typedef typename GeomTraits::FT FT;
+
+    /// @}
+
+  private:
+    using Segment_set_traits = typename std::conditional<
+      std::is_same<typename Traits::Segment_2, Segment_type>::value,
+      internal::Region_growing_traits_2<Traits>,
+      internal::Region_growing_traits_3<Traits> >::type;
+
+    using Point = typename Segment_set_traits::Point;
+    using Segment = typename Segment_set_traits::Segment;
+    using Vector = typename Segment_set_traits::Vector;
+    using Line = typename Segment_set_traits::Line;
+
+    using Squared_length = typename Segment_set_traits::Compute_squared_length;
+    using Squared_distance = typename Segment_set_traits::Compute_squared_distance;
+    using Scalar_product = typename Segment_set_traits::Compute_scalar_product;
+
+  public:
+    /// \name Initialization
+    /// @{
+
+    /*!
+      \brief initializes all internal data structures.
+
+      \tparam NamedParameters
+      a sequence of \ref bgl_namedparameters "Named Parameters"
+
+      \param input_range
+      an instance of `InputRange` with 2D or 3D segments
+
+      \param np
+      a sequence of \ref bgl_namedparameters "Named Parameters"
+      among the ones listed below
+
+      \param segment_map
+      an instance of `SegmentMap` that maps an item from `input_range`
+      to `Kernel::Segment_2` or `Kernel::Segment_3`
+
+      \param traits
+      an instance of `GeomTraits`
+
+      \cgalNamedParamsBegin
+        \cgalParamNBegin{distance_threshold}
+          \cgalParamDescription{the maximum distance from the furthest vertex of a segment to a line}
+          \cgalParamType{`GeomTraits::FT`}
+          \cgalParamDefault{1}
+        \cgalParamNEnd
+        \cgalParamNBegin{angle_deg_threshold}
+          \cgalParamDescription{the maximum accepted angle in degrees between
+          the direction of a segment and the direction of a line}
+          \cgalParamType{`GeomTraits::FT`}
+          \cgalParamDefault{25 degrees}
+        \cgalParamNEnd
+        \cgalParamNBegin{cos_value_threshold}
+          \cgalParamDescription{the cos value computed as `cos(angle_deg_threshold * PI / 180)`,
+          this parameter can be used instead of the `angle_deg_threshold`}
+          \cgalParamType{`GeomTraits::FT`}
+          \cgalParamDefault{`cos(25 * PI / 180)`}
+        \cgalParamNEnd
+        \cgalParamNBegin{min_region_size}
+          \cgalParamDescription{the minimum number of segments a region must have}
+          \cgalParamType{`std::size_t`}
+          \cgalParamDefault{1}
+        \cgalParamNEnd
+      \cgalNamedParamsEnd
+
+      \pre `input_range.size() > 0`
+      \pre `distance_threshold >= 0`
+      \pre `angle_deg_threshold >= 0 && angle_deg_threshold <= 90`
+      \pre `cos_value_threshold >= 0 && cos_value_threshold <= 1`
+      \pre `min_region_size > 0`
+    */
+    template<typename NamedParameters>
+    Least_squares_line_fit_region(
+      const InputRange& input_range,
+      const NamedParameters& np,
```

Comment:
Named parameter must also be the last parameter. Actually the following parameters should be in the named parameter